### PR TITLE
Refactor branch list using src-d/go-git and add GetRefs function

### PR DESCRIPTION
**Pull request by lafriks**
GetRefs will be needed to implement Refs API

Add src-d/go-git dependency and refactored GetBranches to use this lib.

```
Old using git command:
BenchmarkRepository_GetBranches-4           1000           1505930 ns/op
New using src-d/go-git:
BenchmarkRepository_GetBranches-4          10000            150774 ns/op
```

### CODE REVIEW

File: ref.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2018 The Gogs Authors. All rights reserved.
```
Comment (techknowlogick) [inline ref.go]: Gitea

Comment (lafriks) [reply ref.go]: Fixed :)

File: repo_ref.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package git
+
+import (
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"strings"
```
Comment (lunny) [inline repo_ref.go]: import sequences.

Comment (lafriks) [reply repo_ref.go]: Fixed

**Comment by lunny** [comment]
@lafriks the benchmark is for a bare git repository?

**Comment by lafriks** [comment]
@lunny yes

**Comment by lunny** [comment]
It seems we only have a bare repo on testdata

**Comment by lafriks** [comment]
@lunny is that a problem?

**Comment by lunny** [comment]
I prefer pure go library than git command. But for performance measure, I think a bare repo maybe not a good example.

**Comment by lafriks** [comment]
We store all git repos as bare server-side so that is measurement most important for gitea case

**Pull request merged**